rasterio/rio/create.py

```diff
@@ -138,6 +123,17 @@ def create(
                         "Object exists and won't be overwritten without use of the '--overwrite' option."
                     )
 
+    # Prepare the dataset's georeferencing.
+    geo_transform = None
+
+    if bounds:
+        left, bottom, right, top = bounds
+        sx = (right - left) / width
+        sy = (bottom - top) / height
+        geo_transform = Affine.translation(left, top) * Affine.scale(sx, sy)
+    if transform:
+        geo_transform = transform
```

Comment:
I know this might had code complexity but maybe we can add a warning when both transform and bounds are passed? specifically saying which one will be used